### PR TITLE
Backend/issue/518

### PR DESCRIPTION
**Pull request by carltonsmith**
### This pull request changes...

There is an overly permissive unnessisary endpoint for adding users to the system. It needs to be removed

## FUNCTIONALITY
Fixes [#518](https://github.com/raft-tech/TANF-app/issues/518)

Removes create user view mixin from user api view, also removes a serializer which was only used by the no longer necessary create user view.

## TO TEST

**Cloud**
All tests passing shows that the AC is met, as the tests verify the endpoint no longer exists

**Locally**

With the backend running locally run the command `docker-compose run web -c 'pytest'. If all tests pass, nothing has been broken by this PR. A test was also added to make sure the endpoint is no longer there, so all tests passing shows the endpoint no longer exists.

## NOTE
This PR also includes changes from 517, so when that is merged the Files Changed will only show changes from this PR

### This pull request is ready to merge when...

- [x] Meets acceptance criteria for the issue 
- [x] Code is reviewed by someone other than the original author
- [x] Code is tested 
- [x] Experience is approved by UX  
- [x] Meets [a11y checklist](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)
- [x] Meets [Raft's Manual QASP Checklist](https://teams.microsoft.com/l/file/B5285301-13CA-46FD-A723-AFFD578C224A?tenantId=d58addea-5053-4a80-8499-ba4d944910df&fileType=docx&objectUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA%2FShared%20Documents%2FGeneral%2FManual%20Testing%20Practices_Issue173.docx&baseUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA&serviceName=teams&threadId=19:f769bbcb029f4f02b55ae7fad90e310d@thread.skype&groupId=41f194a6-c1d3-4680-933e-c8ee7d17e287):lock: (only applicable for PR to main HHS)
- [x] Documentation is updated 
  - OpenAPI  
  - Readme
  - Entity Relationship Diagram (ERD)

### CODE REVIEW

File: tdrs-backend/tdpservice/users/test/test_api/test_basic.py [hunk 1]
```diff
@@ -40,8 +32,7 @@ def test_cannot_update_user_anonymously(api_client, user):
...
 
 @pytest.mark.django_db
-def test_create_user(api_client, user_data):
-    """Test user creation."""
+def test_create_user_endpoint_not_present(api_client, user_data):
+    """Test removed endpoint is no longer there."""
     response = api_client.post("/v1/users/", user_data)
-    assert response.status_code == status.HTTP_201_CREATED
-    assert User.objects.filter(username=user_data["username"]).exists()
+    assert response.status_code == status.HTTP_404_NOT_FOUND
```
Comment (alexsoble) [inline tdrs-backend/tdpservice/users/test/test_api/test_basic.py:38]: 👍 👍

**Comment by alexsoble** [comment]
# QASP 

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested? — Yes, see changed `test_create_user_endpoint_not_present` test expectation 
+ Are code coverage minimums met?
  + [x] Backend coverage: 100% (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Backend code style checks passing on CircleCI

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

_N/A._

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

_N/A._

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

_N/A, this PR removes a feature/endpoint rather than adding one._

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] OWASP Scan passes on CircleCI

**Comment by alexsoble** [comment]
@lfrohlich, based on the QASP review above, I recommend that you accept and merge this work. This is a feature removal rather than a feature addition -- it removes an endpoint that allows anyone to create a user record. See https://github.com/raft-tech/TANF-app/issues/518 for more details.

**Review by lfrohlich — APPROVED** [review]
Looks good, thanks!

**Pull request merged**